core/src/components/modal/modal.tsx

```diff
@@ -669,6 +669,15 @@ export class Modal implements ComponentInterface, OverlayInterface {
       enteringAnimation.forEach((ani) => ani.destroy());
     }
 
+    if (this.currentBreakpoint) {
+      /**
+       * If the modal is inline and presented again, the visual breakpoint
+       * will not match the current breakpoint. We reset the breakpoint back
+       * to the initial breakpoint on dismiss to resolve this issue.
+       */
+      this.currentBreakpoint = this.initialBreakpoint;
```

Comment:
Should this be set on present instead? It might be confusing to have a modal be dismissed and still have a current breakpoint defined.

Reply:
Good thought 🤔  let me swap that out and test in a dev build.

Reply:
Confirmed with dev-build: `6.1.5-dev.11652119493.129d9329`. Stackblitz is unreliable with recently published dev-builds, but if creating a local clone of a starter app with the stackblitz example template + the dev build, the issue is resolved.